CMakeLists.txt

```diff
@@ -308,8 +308,16 @@ if(WIN32 AND (MINGW OR CYGWIN))
 endif()
 
 if(BUILD_FOR_R)
-    TARGET_LINK_LIBRARIES(lightgbm ${LIBR_CORE_LIBRARY})
+  if(MSVC)
+    set_property(
+      TARGET _lightgbm
+      PROPERTY MSVC_RUNTIME_LIBRARY "MultiThreadedDLL"
+    )
```

Comment:
Maybe it's better to do it here? For the purpose of logical grouping.
https://github.com/microsoft/LightGBM/blob/dbc3d05c9c6f89d296a29b73a85a847c818892ad/CMakeLists.txt#L240-L242

Reply:
yes I think that's a good point. Should I also set it to `MultiThreadedDebugDLL` if `DEBUG` is on?

([here are the docs](https://cmake.org/cmake/help/v3.15/prop_tgt/MSVC_RUNTIME_LIBRARY.html), by the way)

Reply:
ping @guolinke for this question

Reply:
yeah, debug dll is different.

Reply:
BTW, did you check the `/MP` flag in https://github.com/microsoft/LightGBM/blob/master/CMakeLists.txt#L196 ?

Reply:
refer to https://docs.microsoft.com/en-us/cpp/build/reference/md-mt-ld-use-run-time-library?view=vs-2019

Reply:
it seems `MD` is more robust to the environment. Maybe we can only use `MT` for R's dll.

Reply:
@guolinke I'm definitely out of my element here. I'll do whichever you suggest.

Reply:
I actually think we can ignore this completely. Right now the branch doesn't have it, and building the R package with MSVC worked as expected: https://github.com/microsoft/LightGBM/pull/2963#issuecomment-619646274